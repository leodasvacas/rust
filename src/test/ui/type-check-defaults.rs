// Copyright 2017 The Rust Project Developers. See the COPYRIGHT
// file at the top-level directory of this distribution and at
// http://rust-lang.org/COPYRIGHT.
//
// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
// option. This file may not be copied, modified, or distributed
// except according to those terms.
// compile-flags: --error-format=human

use std::iter::FromIterator;

struct Foo<T, U: FromIterator<T>>(T, U);
struct Bar<Z = Foo<i32, i32>>(Z);

struct Qux<A, T=<A as Iterator>::Item>(A, T);

fn main() { }
