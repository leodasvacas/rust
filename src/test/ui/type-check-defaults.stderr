error[E0277]: the trait bound `i32: std::iter::FromIterator<i32>` is not satisfied
  --> $DIR/type-check-defaults.rs:15:12
   |
15 | struct Bar<Z = Foo<i32, i32>>(Z);
   |            ^ a collection of type `i32` cannot be built from an iterator over elements of type `i32`
   |
   = help: the trait `std::iter::FromIterator<i32>` is not implemented for `i32`
   = note: required by `Foo`

error[E0277]: the trait bound `A: std::iter::Iterator` is not satisfied
  --> $DIR/type-check-defaults.rs:17:15
   |
17 | struct Qux<A, T=<A as Iterator>::Item>(A, T);
   |               ^ `A` is not an iterator; maybe try calling `.iter()` or a similar method
   |
   = help: the trait `std::iter::Iterator` is not implemented for `A`
   = help: consider adding a `where A: std::iter::Iterator` bound
   = note: required by `std::iter::Iterator`

error: aborting due to 2 previous errors

